acme_diags/derivations/acme.py

```diff
@@ -304,19 +309,55 @@ def netflux6(rsds, rsus, rlds, rlus, hfls, hfss):
     return var
 
 
-# FIXME: C901 'cosp_bin_sum' is too complex (22)
-def cosp_bin_sum(cld, prs_low0, prs_high0, tau_low0, tau_high0):  # noqa
-    """sum of cosp bins to calculate cloud fraction in specified cloud top pressure and
-    cloud thickness bins, input variable has dimention (cosp_prs,cosp_tau,lat,lon)"""
-    prs = cld.getAxis(0)
-    tau = cld.getAxis(1)
+def adjust_prs_val_units(
+    prs: "FileAxis", prs_val: float, prs_val0: Optional[float]
+) -> float:
+    """Adjust the prs_val units based on the prs.id"""
+    # COSP v2 cosp_pr in units Pa instead of hPa as in v1
+    # COSP v2 cosp_htmisr in units m instead of km as in v1
+    adjust_ids = {"cosp_prs": 100, "cosp_htmsir": 1000}
+
+    if prs_val0:
+        prs_val = prs_val0
+        if prs.id in adjust_ids.keys() and max(prs.getData()) > 1000:
+            prs_val = prs_val * adjust_ids[prs.id]
+
+    return prs_val
+
+
+def determine_cloud_level(
+    prs_low: float,
+    prs_high: float,
+    low_bnds: Tuple[int, int],
+    high_bnds: Tuple[int, int],
+) -> str:
+    """Determines the cloud type based on prs values and the specified boundaries"""
+    # Threshold for cloud top height: high cloud (<440hPa or > 7km), midlevel cloud (440-680hPa, 3-7 km) and low clouds (>680hPa, < 3km)
+    if prs_low in low_bnds and prs_high in high_bnds:
+        return "middle cloud fraction"
```

Comment:
I bumped checking `prs_low` and `prs_high` up first because you need to check both variables simultaneously before moving on to individual checks.

For example, if `prs_low in low_bnds` is first, then it will return immediately if the condition were true. This skips checking `if prs_low in  low_bnds and prs_high in high_bnds` even if this condition were to be true.